ferretdb/ferretdb.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ferretdb
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/FerretDB/FerretDB/internal/clientconn"
+	"github.com/FerretDB/FerretDB/internal/handlers/registry"
+	"github.com/FerretDB/FerretDB/internal/util/logging"
+)
+
+// Config contains a backend connection strings.
+type Config struct {
+	Handler     string
```

Comment:
Should we introduce a new "enum" type for `Handler` (list `pg` and `tigris` as constants of that type etc)?

Reply:
> Should we introduce a new "enum" type for `Handler` (list `pg` and `tigris` as constants of that type etc)?

Let's do it another PR.

Reply:
@seeforschauer then please create an issue or add this item to DoD of the current issue.

Reply:
> @seeforschauer then please create an issue or add this item to DoD of the current issue.

Do we actually need a `enum` for just two handlers?

Reply:
From the design perspective, it's always good to have a more specific type to highlight that this is not an "any" string, but we expect the exact values here.